lib/services/in-memory-db.service.ts

```diff
@@ -133,6 +210,14 @@ export class InMemoryDBService<T extends InMemoryDBEntity> {
     return this.recordMap[id];
   }
 
+  /**
+   * Get a single record of type `T` with the supplied id value as an Observable;
+   * @param id the PK id of the record
+   */
+  public getAsync(id: number): Observable<T> {
+    return of(this.recordMap[id]);
```

Comment:
What are your thoughts on defining the observable on a separate line from returning it? This would allow us to more easily bolt on the `.pipe()` when we start doing things like adding delay operators.

something like:

```suggestion
    const result$ = of(this.get(id));
    return result$;
```

then we could do:

```typescript
    const result$ = of(this.get(id));
    
    if(this.moduleConfig && this.moduleConfig.simulatedDelay) {
     return result$.pipe(delay(this.moduleConfig.simulatedDelay));
    }
    
    return result$;
```

Reply:
Interesting approach. Definitely dig it.